sklearn/tree/_criterion.pyx

```diff
@@ -273,13 +274,12 @@ cdef class ClassificationCriterion(Criterion):
 
     def __dealloc__(self):
         """Destructor."""
-
         free(self.n_classes)
 
     def __reduce__(self):
         return (ClassificationCriterion,
                 (self.n_outputs,
-                 sizet_ptr_to_ndarray(self.n_classes, self.n_outputs)),
+                 sizet_ptr_to_ndarray(self.n_classes, self.n_outputs).copy()),
```

Comment:
because numpy array which is returned by
>sizet_ptr_to_ndarray(self.n_classes, self.n_outputs))
Doesn't own underlying memory (I've added a comment in `sizet_ptr_to_ndarray`), it just uses pointer on values which are stored in C array `self.n_classes`. When you get this tuple from `__repr__`, and object which you got it from gets removed, you can't read this tuple anymore (You can actually, but self.n_classes gets freed, and returned numpy array will show you some garbage, because it still uses pointer to provided C array for him). copy method does deep copy, and resulting array will own underlying memory.

I could alternatively just make `arr = np.zeros(self.n_outputs)`, and copy all values to it. But this approach with `sizet_ptr_to_ndarray().copy()` existed in` _tree.pyx`, so I thought it will be more convenient to use it here too.